include/picongpu/fields/laserProfiles/Polynom.hpp

```diff
@@ -0,0 +1,270 @@
+
+        /* initialize the laser not in the first cell is equal to a negative shift
+         * in time
+         */
+        static constexpr float_X laserTimeShift = Params::initPlaneY * CELL_HEIGHT / SPEED_OF_LIGHT;
+
+        static constexpr float_64 f = SPEED_OF_LIGHT / WAVE_LENGTH;
+
+    };
+} // namespace polynom
+
+namespace acc
+{
+    template< typename T_Unitless >
+    struct Polynom : public T_Unitless
+    {
+        using Unitless = T_Unitless;
+
+        float3_X m_elong;
+        float_X m_phase;
+        typename FieldE::DataBoxType m_dataBoxE;
+        DataSpace< simDim > m_offsetToTotalDomain;
+        DataSpace< simDim > m_superCellToLocalOriginCellOffset;
+
+        /** Device-Side Constructor
+         *
+         * @param superCellToLocalOriginCellOffset local offset in cells to current supercell
+         * @param offsetToTotalDomain offset to origin of global (@todo: total) coordinate system (possibly after transform to centered origin)
+         */
+        HDINLINE Polynom(
+            typename FieldE::DataBoxType const & dataBoxE,
+            DataSpace< simDim > const & superCellToLocalOriginCellOffset,
+            DataSpace< simDim > const & offsetToTotalDomain,
+            float3_X const & elong,
+            float_X const phase
+        ) :
+            m_elong( elong ),
+            m_phase( phase ),
+            m_dataBoxE( dataBoxE ),
+            m_offsetToTotalDomain( offsetToTotalDomain ),
+            m_superCellToLocalOriginCellOffset( superCellToLocalOriginCellOffset )
+        {
+        }
+
+        /** device side manipulation for init plane (transversal)
+         *
+         * @tparam T_Args type of the arguments passed to the user manipulator functor
+         *
+         * @param cellIndexInSuperCell ND cell index in current supercell
+         */
+        template< typename T_Acc >
+        HDINLINE
+        void operator( )(
+            T_Acc const &,
+            DataSpace< simDim > const & cellIndexInSuperCell
+        )
+        {
+            // coordinate system to global simulation as origin
+            DataSpace< simDim > const localCell(
+                cellIndexInSuperCell +
+                m_superCellToLocalOriginCellOffset
+            );
+
+            // transform coordinate system to center of x-z plane of initialization
+            constexpr uint8_t planeNormalDir = 1u;
+            DataSpace< simDim > offsetToCenterOfPlane( m_offsetToTotalDomain );
+            offsetToCenterOfPlane[ planeNormalDir ] = 0; // do not shift origin of plane normal
+            floatD_X const pos = precisionCast< float_X >( localCell + offsetToCenterOfPlane ) * cellSize.shrink< simDim >();
+            // @todo add half-cells via traits::FieldPosition< Solver::NumicalCellType, FieldE >()
+
+            // transversal position only
+            float3_X const w0_3D( Unitless::W0_X, 0., Unitless::W0_Z );
```

Comment:
The naming of these variables differ a bit from the other lasers. In `Unitless::` they are named `W0x` and `W0z`.